### PR TITLE
[RSDK-2619] gpio motor

### PR DESCRIPTION
**Pull request by mattjperez**
- Verified gpio motors work (`ABMotorEsp32` and `PwmDirMotorEsp32`)
- Make`Stoppable` a Super trait for `Base` and `Motor`

### CODE REVIEW

File: src/common/motor.rs [hunk 1]
```diff
@@ -129,16 +134,24 @@ where
...
+    }
     fn set_power(&mut self, pct: f64) -> anyhow::Result<()> {
         info!("setting power to {}", pct);
         self.power = pct;
         Ok(())
     }
-    fn get_position(&mut self) -> anyhow::Result<i32> {
-        Ok(self.pos as i32)
+    fn stop(&mut self) -> anyhow::Result<()> {
+        info!("stopping motor");
```
Comment (npmenard) [inline src/common/motor.rs:152]: make it debug! or remove it (and if you could do the same for the statement in set_power :P)

File: src/esp32/motor.rs [hunk 2]
```diff
@@ -1,3 +1,43 @@
...
+//!         subtype: "motor".to_string(),
+//!         name: "left-motor"
+//!     },
+//!     ResourceType::Motor(Arc::new(Mutex::new(motor))),
+//! );
+//!
+//! let robot_with_motor = LocalRobot(res);
+//!
+//! ```
+//!
```
Comment (npmenard) [inline src/esp32/motor.rs:40]: This is great

**Review by npmenard — CHANGES_REQUESTED** [review]
this looks good, once you fix the CI could you also have Gautham take a look?

**Review by gvaradarajan — APPROVED** [review]
LGTM

**Pull request merged**